themes/base.theme.bash

```diff
@@ -121,6 +121,18 @@ function scm {
 	fi
 }
 
+scm_prompt() {
+	local CHAR=$(scm_char)
+	SCM_PROMPT_BEFORE_CHAR=${SCM_PROMPT_BEFORE_CHAR-"["}
+	SCM_PROMPT_AFTER_INFO=${SCM_PROMPT_AFTER_INFO-"]"}
+
+	if [[ $CHAR = "$SCM_NONE_CHAR" ]]; then
+		return
+	else
+		echo -e "$SCM_PROMPT_BEFORE_CHAR$(scm_char)$SCM_PROMPT_BETWEEN_CHAR_AND_INFO$(scm_prompt_info)$SCM_PROMPT_AFTER_INFO$normal"
```

Comment:
I feel it strange to prepare `SCM_PROMPT_BEFORE_CHAR`, `SCM_PROMPT_BETWEEN_CHAR_AND_INFO`, `SCM_PROMPT_AFTER_INFO` and let the caller partially set something like `SCM_PROMPT_BETWEEN_CHAR_AND_INFO=']['`.

I suggest instead providing a single variable `SCM_PROMPT_FORMAT` such like

```bash
local format=${SCM_PROMPT_FORMAT:-'[%s%s]'}
printf "$SCM_PROMPT_FORMAT\n" "$CHAR" "$(scm_prompt_info)"
```

The caller can set the variable as, e.g.,

```bash
SCM_PROMPT_FORMAT='[%s%s]'
SCM_PROMPT_FORMAT='[%s][%s]'
SCM_PROMPT_FORMAT=' %s (%s)'
```

In this way, it becomes clearer what we are doing.

Reply:
great idea! done

thanks you @akinomyoga for helping me out here :heart: